server/chat-plugins/datasearch.ts

```diff
@@ -1841,6 +1883,7 @@ function runMovesearch(target: string, cmd: string, canAll: boolean, message: st
 	} else {
 		resultsStr += "No moves found.";
 	}
+	if (isTest) return {results, reply: resultsStr};
```

Comment:
It seems to me like this should rather be refactored into two functions, one that returns `results`, and one that transforms `results` into `resultsStr`.

(Not for this PR, but probably a good follow-up PR.)